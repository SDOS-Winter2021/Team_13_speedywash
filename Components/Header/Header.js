/**
 * @module
 */
import React from 'react'
import { Text, View, StyleSheet, Button, TouchableOpacity } from 'react-native'
import { FontAwesome5, MaterialIcons, AntDesign } from '@expo/vector-icons';
import keys from "../../configs/KEYS"

/**
 * Code for footer is present here
 * Three icons clicking on any of them updates the currentView.screen
 * which in return updates whole parent view
 * @param {Object} obj - An object
 * @param {module:globaltypes.VisibleView} obj.currentView - current visible/selected view
 * @param {function} obj.setcurrentView - current view setter
*/
function Header({ currentView, setcurrentView }) {
    return (
        <View style={styles.HeaderStyle}>
            <Text style={styles.headerTitle}>Speedywash India</Text>
            <View style={{ flex: 0.17, flexDirection: "row" }}>
                <TouchableOpacity style={styles.icons} onPress={() => setcurrentView({ screen: keys.screens.CART, header: true, footer: true })}>
                    <FontAwesome5 name='shopping-cart' size={32} color={'white'} />
                </TouchableOpacity>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    HeaderStyle: {
        // flex: 1,
        height: '100%',
        backgroundColor: keys.colors.MAIN,
        flexDirection: "row"
    },
    headerTitle: {
        fontSize: 20,
        color: 'white',
        alignSelf: 'center',
        margin: '5%',
        flex: 0.83
    },
    icons: {
        alignSelf: 'center'
    }
})

export default Header;