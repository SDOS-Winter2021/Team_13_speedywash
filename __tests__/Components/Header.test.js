import React from 'react';
import renderer from 'react-test-renderer';
import Header from "../../Components/Header/Header"
import {test_user} from "../global"

it('Header renders correctly', () => {
    const tree = renderer.create(<Header currentView={{screen: "HOMEPAGE"}} />).toJSON();
    expect(tree).toMatchSnapshot();
})

