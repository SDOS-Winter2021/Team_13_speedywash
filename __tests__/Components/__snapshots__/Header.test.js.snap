// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#006BAD",
      "flexDirection": "row",
      "height": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "white",
        "flex": 0.83,
        "fontSize": 20,
        "margin": "5%",
      }
    }
  >
    Speedywash India
  </Text>
  <View
    style={
      Object {
        "flex": 0.17,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
